pkg/synchromanager/clustersynchro_manager.go

```diff
@@ -183,6 +188,12 @@ func (manager *Manager) enqueue(obj interface{}) {
 		return
 	}
 
+	cluster := obj.(*clusterv1alpha2.PediaCluster)
+	if manager.shardingName != cluster.Spec.ShardingName &&
+		(cluster.Status.ShardingName == nil || manager.shardingName != *cluster.Status.ShardingName) {
+		return
+	}
```

Comment:
Missing filter `spec.sharding == name and status.sharding != nil and status.sharding != name`